### ISSUE TITLE
Default price faceted search does not work with 1.6

### ISSUE THREAD

**Issue by petedermott** [django-oscar/django-oscar#2739]
### Issue Summary

The price based faceted search (http://django-oscar.readthedocs.io/en/releases-1.1/ref/settings.html#oscar-search-facets) does not work with the current sandbox site 

### Steps to Reproduce

1. Follow the "Running Sandbox locally" guide at http://django-oscar.readthedocs.io/en/releases-1.1/internals/sandbox.html#run-the-sandbox-locally
2. Update the settings.py file with a solr or elasticsearch 2.4 haystack setting
3. Start the runserver, browse to a category and try to use the filters.

I understand that there is work afoot to switch to our own ES backed search engine but at the moment these filters just do not work as advertised

### Technical details

* Python version: 3.6.5
* Django version: 2.0.6.
* Oscar version: 2.0.dev0

**Comment by solarissmoke** [comment]
You're right, this is broken. I've tested on Elasticsearch, and the issue there is that Elasticsearch 2 [dropped support for range facets](https://www.elastic.co/guide/en/elasticsearch/reference/2.2/search-facets-range-facet.html). Haystack in turn simply dropped all support for range facets from its Elasticsearch 2 backend.

I don't know what the issue is on Solr, but I suspect it's a similar problem because recent versions of Solr handle range facets differently.

This highlights the problem with Haystack and the motivation to drop it from Oscar. Unfortunately I don't see us being able to fix this quickly as the issue is with Haystack itself, which hasn't kept up to date with changes to Elasticsearch and Solr.

We are working on a new search backend (using Elasticsearch 5+) - see #2734 - which I think is the best long term solution. I don't really have any good suggestions in the mean time though.
